ably/rest_client_test.go

```diff
@@ -270,11 +270,56 @@ func TestRSC7(t *testing.T) {
 			t.Errorf("expected %s got %s", ably.AblyVersion, h)
 		}
 	})
-	t.Run("must set lib header", func(t *testing.T) {
-		h := req.Header.Get(ably.AblyLibHeader)
-		if h != ably.LibraryString {
-			t.Errorf("expected %s got %s", ably.LibraryString, h)
+}
+
+func TestRest_RSC7_AblyAgent(t *testing.T) {
+	t.Parallel()
+
+	app, err := ablytest.NewSandbox(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer app.Close()
+
+	t.Run("RSC7d2 : Should set fallback host header", func(t *testing.T) {
+		t.Parallel()
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusInternalServerError)
+		}))
+		defer server.Close()
+
+		opts := []ably.ClientOption{
+			ably.WithEnvironment(ablytest.Environment),
+			ably.WithTLS(false),
+			ably.WithUseTokenAuth(true),
 		}
+		// set up the proxy to forward all requests except a specific fallback to the server,
+		// whilst that fallback goes to the regular endpoint
+		serverURL, _ := url.Parse(server.URL)
+
+		var agentHeaderValue string
+		proxy := func(r *http.Request) (*url.URL, error) {
+			agentHeaderValue = r.Header.Get(ably.AblyAgentHeader)
+			return serverURL, nil
+		}
+
+		opts = append(opts, ably.WithHTTPClient(&http.Client{
+			Transport: &http.Transport{
+				Proxy:        proxy,
```

Comment:
Why does this test need to use a proxy?

Reply:
Yes, I strongly remember one of the issue, where I could only read headers as a part of proxy. Well, its working in realtime test, there is no point to keep the same in rest tests